nio_cli/commands/blockcheck.py

```diff
@@ -0,0 +1,178 @@
+        self.all_contents = os.listdir('.')
+        self.block_files = [
+            f for f in self.all_contents if 'block.py' in f and 'base' not in f
+        ]
+        self.blocks_in_spec = []
+        self.spec_versions_dict = {}
+        self.release_versions_dict = {}
+
+    def run(self):
+        # Questions;
+        # OK to assume each block has a corresponding file and each file only has one block?
+        # OK that everything uses `print` statements?
+            # How can I test this?
+                # Switch to having check_x methods return a string, then print that string?
+        # Ok that all checks are built off the spec (self.blocks_in_spec)?
+            # Would it be better to base checks off the block file(s)?
+
+        self.print_check('PEP8')
+        self.check_pep8()
+
+        self.print_check('spec.json')
+        # use `sys.exit()` for "failures" in check_spec() because
+        # the checks after check_spec all use self.blocks_in_spec
+        self.check_spec()
+
+        self.print_check('README.md')
+        self.check_readme()
+
+        self.print_check('release.json')
+        self.check_release()
+
+        # TODO: Check that block file version is the same as spec/release
+        self.print_check('version')
+        self.check_version()
+
+        # What should I check for in regards to naming specifications?
+        self.print_check('class and file name')
+        self.check_naming()
+
+    def check_pep8(self):
+
+        shell_pep8 = 'pep8 .'
+        subprocess.call(shell_pep8, shell=True)
+        print('')
+        # Any way to grab pep8 output?
+
+    def check_spec(self):
+
+        if os.path.exists('spec.json'):
+            with open('spec.json') as f:
+                spec_dict = json.load(f)
+                self.blocks_in_spec = [
+                    k.split('/')[1] for k, v in spec_dict.items()
+                ]
+
+            if len(self.blocks_in_spec) > len(self.block_files):
+                print('There\'s extra blocks in the spec file')
+                sys.exit()
+            if len(self.blocks_in_spec) < len(self.block_files):
+                print('Not all blocks are in the spec file')
+                sys.exit()
+
+            for block in self.blocks_in_spec:
+                keys = ['version', 'description', 'properties', 'inputs',
+                        'outputs', 'commands']
+                for key in keys:
+                    if key not in [k for k in spec_dict['nio/' + block]]:
+                        print('{} block is missing {}'.format(block, key))
+                        sys.exit()
+                self.spec_versions_dict[block] = \
+                    spec_dict['nio/' + block]['version']
+
+            for key in ['version', 'description', 'properties']:
+                if not spec_dict['nio/' + block][key] \
+                        or spec_dict['nio/' + block][key] == '':
+                    print('Please fill in the {} of the {} block'.format(
```

Comment:
Maybe instead of being extremely polite, just say "missing X in X block"? :) I just imagine this output would get a bit grating if there was a lot of stuff wrong. Just my opinion. This goes for the other print statements as well.